src/pycell.rs

```diff
@@ -301,7 +303,11 @@ impl<T: PyClass> PyCell<T> {
     /// Panics if the value is currently mutably borrowed. For a non-panicking variant, use
     /// [`try_borrow`](#method.try_borrow).
     pub fn borrow(&self) -> PyRef<'_, T> {
-        self.try_borrow().expect("Already mutably borrowed")
+        Self::borrow_bound(&self.as_borrowed())
+    }
+
+    pub(crate) fn borrow_bound<'py>(obj: &Bound<'py, T>) -> PyRef<'py, T> {
+        Self::try_borrow_bound(obj).expect("Already mutably borrowed")
     }
```

Comment:
Since these are essentially constructors for `PyRef` and `PyRefMut` respectively, maybe it makes sense to move them to their respective `impl` block instead of implementing them on `PyCell`

Reply:
Good idea, will do that 👍